### PR TITLE
[RNMobile] Add API fetch test helper

### PR DESCRIPTION
**Pull request by fluiddot**
## What?

Adds a test helper to allow mocking API fetch requests in tests.

## Why?

This helper will simplify the test logic for cases blocks/components make API requests.

## How?

A new helper has been added that accepts an array of objects with the following parameters:
- `request`: This object contains the parameters that we'll use to match requests. The parameters are the same as we pass to the `apiFetch` function.
- `response`: The result we want to return to requests that match the parameters set above.

Additionally, some tests of the Image block have been updated to use this helper. This block fetches media data from the API when is mounted, so it was a good example to use the helper.

## Testing Instructions

1. Run the command: `npm run native test -- image/test/edit.native`
2. Observe that all tests pass.
3. Observe that `Unit Tests / Mobile` CI jobs pass.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

File: test/native/integration-test-helpers/setup-api-fetch.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	apiFetch.mockImplementation( async ( options ) => {
+		const matchedResponse = responses.find(
+			( { request: { path, method } } ) => {
+				return (
+					path === options.path &&
+					( ! options.method || method === options.method )
+				);
+			}
+		);
+		return matchedResponse?.response;
```
Comment (SiobhyB) [inline test/native/integration-test-helpers/setup-api-fetch.js:48]: Would there be a benefit to logging an error here if there is no `matchedResponse`?

```
		if ( ! matchedResponse ) {
			throw new Error( `No response for path: ${ path }` );
		}

		return matchedResponse.response;
```

**Review by SiobhyB — APPROVED** [review]
LGTM! The tests pass as expected and all looks good. I just asked one question, but nothing blocking. Thanks @fluiddot! 🙌

File: test/native/integration-test-helpers/setup-api-fetch.js (hunk 1 above)
Comment (fluiddot) [reply test/native/integration-test-helpers/setup-api-fetch.js:48]: Actually, I originally added this condition but noticed the editor makes a set of requests always upon its initialization. This made me think that probably makes more sense to delegate checking what requests should be made or shouldn't to the test logic, instead here. In any case, we could add an option to make this function more restrictive and fail the test if an unexpected request is made. Maybe it's useful in some cases 🤔 .

For reference, the following requests are made when the editor is initialized:
- `/wp/v2/settings`
- `/wp/v2/media`
- `/wp/v2/pages`
- `/wp/v2/block-patterns/patterns`
- `/wp/v2/block-patterns/categories`
- `/wp/v2/posts/post-id-<POST_GUID>`

Comment (SiobhyB) [reply test/native/integration-test-helpers/setup-api-fetch.js:48]: Thank you for sharing the logic behind that decision, that makes sense to me.

> In any case, we could add an option to make this function more restrictive and fail the test if an unexpected request is made. Maybe it's useful in some cases 🤔 .

I trust your judgement, but have a sense that the value would be relatively low. The current code looks good to me :) thanks again for expanding on the logic here.

Comment (fluiddot) [reply test/native/integration-test-helpers/setup-api-fetch.js:48]: > I trust your judgement, but have a sense that the value would be relatively low. The current code looks good to me :) thanks again for expanding on the logic here.

Ok, I'll merge the PR as-is, thanks!

**Pull request merged**